app/components/agenda/agendaitem/send-to-vp-modal.js

```diff
@@ -38,20 +51,104 @@ export default class SendToVpModalComponent extends Component {
 
-      const agendaitemPieces = await latestAgendaitem.pieces;
-      let pieces = report ? [report, ...agendaitemPieces] : agendaitemPieces;
-      for (const piece of pieces) {
+      const DECREET = await this.store.findRecordByUri(
+        'concept',
+        CONSTANTS.DOCUMENT_TYPES.DECREET
+      );
+      const MEMORIE = await this.store.findRecordByUri(
+        'concept',
+        CONSTANTS.DOCUMENT_TYPES.MEMORIE
+      );
+      const NOTA = await this.store.findRecordByUri(
+        'concept',
+        CONSTANTS.DOCUMENT_TYPES.NOTA
+      );
+      const ADVIES = await this.store.findRecordByUri(
+        'concept',
+        CONSTANTS.DOCUMENT_TYPES.ADVIES
+      );
+      const INTERN_OVERHEID = await this.store.findRecordByUri(
+        'concept',
+        CONSTANTS.ACCESS_LEVELS.INTERN_OVERHEID
+      );
+      const PUBLIEK = await this.store.findRecordByUri(
+        'concept',
+        CONSTANTS.ACCESS_LEVELS.PUBLIEK
+      );
+      const subcasePieces = (await this.store.queryAll('piece', {
+        'filter[document-container][type][:id:]': [
+          DECREET.id,
+          MEMORIE.id,
+          NOTA.id,
+          ADVIES.id,
+        ].join(','),
+        'filter[access-level][:id:]': [
+          INTERN_OVERHEID.id,
+          PUBLIEK.id,
+        ].join(','),
+        'filter[submission-activity][subcase][:id:]': subcase.id,
+        sort: 'name',
+      })).slice();
+
+      const allPieces = report ? [report, ...subcasePieces] : subcasePieces;
+      const piecesReadyToBeSent = [];
+
+      for (const piece of allPieces) {
         const documentContainer = await piece.documentContainer;
         await documentContainer?.type;
         await piece.signedPiece;
-        await piece.file;
+        const file = await piece.file;
+        await file.derived;
       }
       const subcaseType = await subcaseTypePromise;
+
+      for (const piece of allPieces) {
+        const submittedPieces = (await piece.submittedPieces).slice();
+        if (submittedPieces.length === 0) {
+          piecesReadyToBeSent.push(piece);
+          continue;
+        }
+
+        let hasSentPdf = false;
+        let hasSentWord = false;
+        let hasSentSigned = false;
+        for (const submittedPiece of submittedPieces)  {
+          const file = await piece.file;
+          const derived = await file.derived;
+          let pdf;
+          let word;
+          let signed;
+          if (derived) {
+            pdf = derived;
+            word = file;
+          } else {
+            pdf = file;
+          }
+          const signedPieceCopy = await piece.signedPieceCopy;
+          signed = await signedPieceCopy?.file;
+
+          if (submittedPiece.unsignedFile === pdf.uri) {
+            hasSentPdf = true;
+          }
+          if (submittedPiece.wordFile === word?.uri) {
+            hasSentWord = true;
+          }
+          if (submittedPiece.signedFile === signed?.uri) {
+            hasSentSigned = true;
+          }
```

Comment:
I'm always reluctant to write something like `if (somestatement) { someBoolean = true } else { someBoolean = false }` unless it's inside a loop where you only set the value of `someBoolean` once and then break the loop.

These statements can be rewritten as `someBoolean = someStatement`, so in this case, I'd prefer

```
let hasSentPdf = (submittedPiece.unsignedFile === pdf.uri);
let hasSentWord = (submittedPiece.wordFile === word?.uri);
let hasSentSigned = (submittedPiece.signedFile === signed?.uri);
```

which replaces 12 lines of code by 3.

Reply:
The problem is that we only want to assign in the positive case.

When we submit a piece we only send what hasn't been sent. Take the following case:

- Piece A has PDF and DOCX, we send both to VP
- Piece A gets signed
- Signed version of Piece A gets sent to VP, original PDF and DOCX won't be sent because they haven't changed
- We want to send to VP again and now calculate which pieces have been sent (done in this loop)

In the current implementation, in the first iteration we'll get the submitted piece with PDF and DOCX, mark `hasSentPdf` and `hasSentWord` as true. In the next iteration we'll get the signed piece and mark `hasSentSigned`  as true. Once done, we know all three have been sent and that the piece no longer needs to be sent to the VP.

If we assign like you suggest, in the second iteration we'll find:

```js
let hasSentPdf = (submittedPiece.unsignedFile === pdf.uri); // False, because 2nd submitted piece has no unsignedFile
let hasSentWord = (submittedPiece.wordFile === word?.uri); // False, because 2nd submitted piece has no wordFile
let hasSentSigned = (submittedPiece.signedFile === signed?.uri); // True
```

Which will cause the piece to be erroneously resent.